Reduce bundle size dynamic case detail (human test needed)

yshing opened the pull request:
- upgrade gatsby
- use /cases page filter to handle /cases/xxx
- no to build cases/xxx
- tests needed

yshing committed:
Reduce size for generated cases/index.html and remove source map from production build

yshing committed:
upgrade-gatsby

yshing committed:
Use useRouter over custom reducer for paths

yshing committed:
Create wildcard page for cases

yshing committed:
make /cases/* as case number filter

yshing committed:
SEO and inject sitemap

yshing committed:
Merge remote-tracking branch 'origin/master' into reduce-bundle-size-dynamic-case-detail

# Conflicts:
#	gatsby-node.js

nandiheath commented on gatsby-node.js:
```diff
@@ -722,44 +726,23 @@ exports.createPages = async ({ graphql, actions }) => {
   return Promise.resolve(null)
 }
+
+exports.onPostBuild = ({ reporter }) => {
```
good catch!

nandiheath commented on src/components/molecules/TagStyleFilter.js:
```diff
@@ -210,6 +211,7 @@ const TagStyledFilter = props => {
   const resetFilters = () => {
     applyFilter(initialFiltersState)
   }
+  useEffect(() => applyFilter(filters), [onListFiltered])
```
eslint errors here:
 `React Hook useEffect has missing dependencies: 'applyFilter' and 'filters'. Either include them or remove the dependency array`

nandiheath reviewed (changes requested):
Yeah overall looks good
especially thank you so much for rewriting the context store with hooks, great job!

just some minor eslint warning needs to be fixed

yshing committed:
Update TagStyleFilter.js

nandiheath committed:
Update gatsby-node.js

nandiheath committed:
Update TagStyleFilter.js

i assumed it should be called whenever `filters` are updated. correct me if this is wrong.
